src/rvsdg/optimize_direct_jumps.rs

```diff
@@ -19,10 +19,68 @@ use crate::Optimizer;
 
 impl SimpleCfgFunction {
     pub fn optimize_jumps(&self) -> Self {
-        self.fuse_down().collapse_empty_blocks()
+        // fusing down only needs to happen once
+        // fuze up may need to run until fixed point
+        // collapse empty blocks may also need to run until fixed point
+        let mut res = self
+            .fuse_down()
+            .fuze_up()
```

Comment:
why do we need both fuze_up and fuze_down?

Reply:
fuzing could happen multiple times- see comment above
fuze_up is aware of `id` instructions so it can do things where a block as multiple parents
fuze_down only handles direct jumps, it's much simpler